framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/scenario/WorkbenchKieServerPersistentScenarioImpl.java

```diff
@@ -89,24 +94,29 @@ protected void deployCustomResource() {
             gitProvider = Git.createProvider(project, request.getGitSettings());
         }
 
-        registerTrustedSecret(kieApp.getSpec().getObjects().getConsole());
-
+        if (!isCustomTrustedSecretRegistered(kieApp.getSpec().getObjects().getConsole())) {
+            registerTrustedSecret(kieApp.getSpec().getObjects().getConsole());
+        }
         for (Server server : kieApp.getSpec().getObjects().getServers()) {
-            registerTrustedSecret(server);
+            if (!isCustomTrustedSecretRegistered(server)) {
+                registerTrustedSecret(server);
+            }
```

Comment:
Why do we need to check whether the ssl is configured? The values will remain the same, so it should not matter if we set the environment variables again and they won't change in the second execution, right?

Reply:
Yes, the value remain same but once when ssl is configured, I think there is no need to creating it again. If there is existing configuration, we should use it. It's also closer to manual update, when is KieApp updated manually we do not created the custom secret again.

Reply:
Ok, I see the problem. The "env" is a field, if we don't check this, we'll be adding the same value several times. 
Maybe in order to avoid having the same hard coded values in difference places (example: "HTTPS_NAME" or "HTTPS_PASSWORD"), what about adding a transient method in the Server.java to add the env if it does not exist?

```java
    @Transient
    public void addEnvIfNotSet(String name, String value) {
        if (Stream.of(getEnv()).map(Env::getName).noneMatch(envName -> StringUtils.equals(envName, name))) {
            addEnv(new Env(name, value));
        }
    }
```

Then, it would be only about to replace the existing "addEnv" method to "addEnvIfNotSet" in OpenShiftOperatorScenario. We would not need to worry about having new methods to check existing envs or deal with duplicated hardcoded values.
What do you think?